Django 1.10 support

gavinwahl opened the pull request.

gavinwahl committed:
Django 1.10 support

iainwatts commented on widgy/models/base.py:
```diff
@@ -839,13 +839,19 @@ def clone(self):
+        except AttributeError:
+            pass
         new.save()
+        print((self, self.pk), (new, new.pk))
```
Remove print statement

iainwatts commented on widgy/models/base.py:
```diff
@@ -839,13 +839,19 @@ def clone(self):
         # See https://code.djangoproject.com/ticket/4027
         new = copy.copy(self)
         unset_pks(new)
+        try:
```
Add comment to explain what this does?

gavinwahl committed:
Flatten the context into a dict using it for render_to_string

render_to_string is documented to take a dict, not a Context object.
Normally this doesn't matter, but django-compressor breaks when we do it
(it calls .flatten on the context, which is now nested, and that causes
an exception)

Commit:
Create missing migration for widgypage

Commit:
Update quickstart docs for Django 1.10

Commit:
Fix related lookup JS bug

We need to attach events handlers for popups when loading a page via
ajax.

Commit:
Fix large file upload bug

When large files are uploaded, they are stored as temp files and moved
when the storage.save method is called. This method was called in
FileUpload's serialize_value method, so if serialize_value was called
more than once (ex: by the EmailSuccessHandler), the temp file would
be moved in the first call, and would not exist for subsequent calls,
resulting in an IOError.

Commit:
Set Forms admin page title to Form Submissions

lockie commented:
It would be lovely to have this merged. Anyone?

Commit:
Modify tox to install 1.10 on 1.10 test

Commit:
Pin version of Py

Tox on my machine and on Travis is installing py version 1.5, which is
too new to be used with pytest. We can't reproduce this on other
machines reliably, so we have no idea what's going on.

Pinning Py to version 1.4.34 seems to resolve the issue.

gavinwahl committed:
Use the builtin UUIDField

Django extensions removed theirs now that all supported versions of
django have it. The tests still aren't passing with this commit.

Commit:
Update form_builder to use built-in UUID field

This field was added in Django 1.8

gavinwahl committed:
Explicitly cast to uuid for postgres

gavinwahl committed:
fix BeautifulSoup warning

gavinwahl committed:
explicitly require html5lib (see 7c42715648f6257)

Commit:
Switch to using connection vendor

Commit:
Attempt to fix travis mysql UTF8 handling

gavinwahl committed:
Remove tests for unsupported versions

- Django 1.9 is not supported
- The latest versions of django do not support python 3.3

Commit:
Attempt to change default mysql encoding in travis

Commit:
Change default character encoding for mysql in travis

Commit:
Print node version in travis

Commit:
Use node 8 to run tests

Commit:
Attempt to get travis to use latest node LTS

Commit:
specify language as node for javascript tests

Commit:
Prevent tox from being installed on javascript tests

Commit:
Fix data loss error

We accidentally created a migration that truncates UUIDs to 32
characters. This can cause data loss as before we were storing them as
36 character long strings

gavinwahl committed:
Django 1.11 support

gavinwahl committed:
uuid identifier fields should not be editable

gavinwahl committed:
Support ManifestStaticFilesStorage

It doesn't work to call static with a partial url. We'll just have to
use the STATIC_URL settings.

gavinwahl committed:
correctly migrate Form.ident to uuid

gavinwahl committed:
Improve support for loading CKEditor from a script tag

CKEDITOR_BASEPATH can't be changed after ckeditor.js is loaded, but
CKEDITOR.basePath can. This helps if you are using
ManifestStaticFileStorage, because it changes the name of ckeditor.js.

gavinwahl committed:
Remove ie8 support that references nonexistent files

gavinwahl committed:
This is the future

gavinwahl committed:
The container .formField is not always the direct parent of the input

Specifically, with filer file fields.

santoshray02 commented:
I am getting this error

django.db.utils.OperationalError: no such table: django_content_type
